https://bleacherreport.com/articles/113144-cracking-the-code-how-to-calculate-hollingers-per-without-all-the-mess

We used the above website to calculate [Hollinger's PER](https://en.wikipedia.org/wiki/Player_efficiency_rating#Calculation)

Because we are estimating Hollinger's PER using the above estimate, we want the difference between our calculated PER values vs. online PER values to be within a margin of error of 2 points.

In [22]:
import pandas as pd

df = pd.read_csv('NBA_STATS_WITH_PER.csv')
df


Unnamed: 0                   ID    Year             Player  Pos   Age  \
0               0    Curly Armstrong_1  1950.0    Curly Armstrong  G-F  31.0   
1               1       Cliff Barker_1  1950.0       Cliff Barker   SG  29.0   
2               2      Leo Barnhorst_1  1950.0      Leo Barnhorst   SF  25.0   
3               3         Ed Bartels_1  1950.0         Ed Bartels    F  24.0   
4               4         Ed Bartels_2  1950.0         Ed Bartels    F  24.0   
...           ...                  ...     ...                ...  ...   ...   
24619       24619        Cody Zeller_4  2017.0        Cody Zeller   PF  24.0   
24620       24620       Tyler Zeller_5  2017.0       Tyler Zeller    C  27.0   
24621       24621  Stephen Zimmerman_1  2017.0  Stephen Zimmerman    C  20.0   
24622       24622        Paul Zipser_1  2017.0        Paul Zipser   SF  22.0   
24623       24623        Ivica Zubac_1  2017.0        Ivica Zubac    C  19.0   

        Tm     G    GS      MP  ...    ORB    DRB    TRB    AST   STL   BLK  \
0      FTW  63.0   NaN     NaN  ...    NaN    NaN    NaN  176.0   0.0   0.0   
1      INO  49.0   NaN     NaN  ...    NaN    NaN    NaN  109.0   0.0   0.0   
2      CHS  67.0   NaN     NaN  ...    NaN    NaN    NaN  140.0   0.0   0.0   
3      TOT  15.0   NaN     NaN  ...    NaN    NaN    NaN   20.0   0.0   0.0   
4      DNN  13.0   NaN     NaN  ...    NaN    NaN    NaN   20.0   0.0   0.0   
...    ...   ...   ...     ...  ...    ...    ...    ...    ...   ...   ...   
24619  CHO  62.0  58.0  1725.0  ...  135.0  270.0  405.0   99.0  62.0  58.0   
24620  BOS  51.0   5.0   525.0  ...   43.0   81.0  124.0   42.0   7.0  21.0   
24621  ORL  19.0   0.0   108.0  ...   11.0   24.0   35.0    4.0   2.0   5.0   
24622  CHI  44.0  18.0   843.0  ...   15.0  110.0  125.0   36.0  15.0  16.0   
24623  LAL  38.0  11.0   609.0  ...   41.0  118.0  159.0   30.0  14.0  33.0   

        TOV     PF    PTS        PER  
0       0.0  217.0  458.0  15.000000  
1       0.0   99.0  279.0  15.000000  
2       0.0  192.0  438.0  15.000000  
3       0.0   29.0   63.0  15.000000  
4       0.0   27.0   59.0  15.000000  
...     ...    ...    ...        ...  
24619  65.0  189.0  639.0  17.863066  
24620  20.0   61.0  178.0  14.595065  
24621   3.0   17.0   23.0   8.419667  
24622  40.0   78.0  240.0   7.972956  
24623  30.0   66.0  284.0  18.510399  

[24624 rows x 33 columns]

In [31]:
#check players PER properly calculated

import random

seed_val = 225 #used to select random player

for i in range(5):
    random.seed(seed_val)
    index = random.randint(0, len(df))
    #print(index)
    player = df.loc[index]
    print( i+1, " Name ",player['Player'], "Team ",player['Tm'], " Year ",player['Year']," PER ", player['PER'])
    seed_val+=1

1  Name  Troy Daniels Team  MIN  Year  2015.0  PER  7.782025974025973
2  Name  Bryon Russell Team  UTA  Year  2002.0  PER  9.754204704704708
3  Name  Antonio Lang Team  CLE  Year  1997.0  PER  7.324786476868327
4  Name  Reggie Williams Team  OKC  Year  2014.0  PER  16.950176470588236
5  Name  Maciej Lampe Team  TOT  Year  2005.0  PER  6.979015789473686


1. Troy Daniels- Calculated PER = 7.782025974025973  , Actual PER = 7.4
https://www.basketball-reference.com/players/d/danietr01.html#advanced

2. Bryon Russell- Calculated PER = 9.754204704704708  , Actual PER = 9.8
https://www.basketball-reference.com/players/r/russebr01.html#advanced

3. Antonio Lang - Calculated PER = 7.324786476868327  , Actual PER = 8.0
https://www.basketball-reference.com/players/l/langan02.html#advanced

4. Reggie Williams- Calculated PER = 16.950176470588236  , Actual PER = 16.4
https://www.basketball-reference.com/players/w/willire02.html#advanced

5. Maciej Lampe- Calculated PER = 6.979015789473686  , Actual PER = 7.3
https://www.basketball-reference.com/players/l/lampema01.html#advanced


As demonstrated above for our random sample, the difference between the actual vs estimated PERs fall within our margin of error: We can reasonably declare that our calculation of PER was accurate, and we were able to properly scrape relevant player data.